sklearn/utils/estimator_checks.py

```diff
@@ -380,6 +381,27 @@ def check_estimator_sparse_data(name, Estimator):
 
 
 @ignore_warnings(category=(DeprecationWarning, UserWarning))
+def check_sample_weights_pandas_series(name, Estimator):
+    # check that estimators will accept a 'sample_weight' parameter of
+    # type pandas.Series in the 'fit' function.
+    estimator = Estimator()
+    if has_fit_parameter(estimator, "sample_weight"):
+        try:
+            import pandas as pd
+            X = pd.DataFrame([[1, 1], [1, 2], [1, 3], [2, 1], [2, 2], [2, 3]])
+            y = pd.Series([1, 1, 1, 2, 2, 2])
+            weights = pd.Series([1] * 6)
+            try:
+                estimator.fit(X, y, sample_weight=weights)
+            except ValueError:
+                raise ValueError("Estimator {0} raises error if "
+                                 "'sample_weight' parameter is type "
+                                 "{1}".format(name, pd.Series))
+        except ImportError:
+            pass
```

Comment:
Maybe we should raise a ``SkipTest("Pandas not installed, not testing pandas series as class weight")``? That would be more explicit.